libs/hdf-converters/src/mappings/ChecklistMappingData.ts

```diff
@@ -0,0 +1,241 @@
+   */
+  uniqueId: string;
+  /** The filename that this file was uploaded under. */
+  filename: string;
+
+  database_id?: string;
+
+  asset: ChecklistAsset;
+  stigs: ChecklistStig[];
+  raw: Record<string, unknown>;
+};
+
+type ChecklistAsset = {
+  role: string;
+  assettype: string;
+  hostname: string;
+  hostip: string;
+  hostmac: string;
+  hostfqdn: string;
+  marking: string;
+  targetcomment: string;
+  techarea: string;
+  targetkey: string;
+  webordatabase: boolean;
+  webdbsite: string;
+  webdbinstance: string;
+};
+
+export type ChecklistStig = {
+  header: StigHeader;
+  vulns: ChecklistVuln[];
+};
+
+type StigHeader = {
+  version: string;
+  classification: string;
+  customname?: string;
+  stigid: string;
+  description: string;
+  filename: string;
+  releaseinfo?: string;
+  title: string;
+  uuid: string;
+  notice?: string;
+  source?: string;
+};
+
+type ChecklistVuln = {
+  status: string | undefined; // Look into alternate way of mapping.  Shouldn't be undefined
+  findingDetails: string;
+  comments: string;
+  severityOverride: string;
+  severityJustification: string;
+  vulnNum: string;
+  severity: string;
+  groupTitle: string;
+  ruleId: string;
+  ruleVersion: string;
+  ruleTitle: string;
+  vulnDiscuss: string;
+  iaControls: string;
+  checkContent: string;
+  fixText: string;
+  falsePositives: string;
+  falseNegatives: string;
+  documentable: string;
+  mitigations: string;
+  potentialImpact: string;
+  thirdPartyTools: string;
+  mitigationControl: string;
+  responsibility: string;
+  securityOverrideGuidance: string;
+  checkContentRef: string;
+  weight: string;
+  class: string;
+  stigRef: string;
+  targetKey: string;
+  stigUuid: string;
+  legacyId: string;
+  cciRef: string;
+};
+
+/**
+ * Gets attribute data
+ * @param stigdata - array of raw stig data json
+ * @param tag - attribute name
+ * @returns attribute data
+ */
+function getAttributeData(stigdata: unknown[], tag: string): string {
+  const results = stigdata.filter((attribute: unknown) => {
+    return _.get(attribute, 'vulnattribute') === tag;
+  });
+  if (results.length === 1) {
+    return _.get(results[0], 'attributedata') as unknown as string;
+  } else {
+    return results
+      .map((result: unknown) => _.get(result, 'attributedata'))
+      .join('; ');
+  }
+}
+
+/**
+ * Gets si data
+ * @param stiginfo - array of raw stig information json
+ * @param tag - attribute name
+ * @returns si data
+ */
+function getSiData(stiginfo: unknown[], tag: string): string {
+  const results = stiginfo.filter((attribute: unknown) => {
+    return _.get(attribute, 'sidname') === tag;
+  });
+  if (results.length === 1) {
+    return _.get(results[0], 'siddata') as unknown as string;
+  } else {
+    return results
+      .map((result: unknown) => _.get(result, 'siddata'))
+      .join('; ');
+  }
+}
+
+// Status mapping for going to and from checklist
+const STATUS_MAPPING: Map<string | undefined, string> = new Map([
```

Comment:
make this only compatible with ckl2hdf - we'll just create the inverse mapping when we do the reverse mapper

i think life would be a lot easier with bracket notation instead of .gets so consider converting it into an object as well instead of using a map

Reply:
Using enum instead of map now